### PR TITLE
Prune read only properties from input & output

### PR DESCRIPTION
**Pull request by anshikg**
*Issue #, if available:*

*Description of changes:* prune read only properties from both input to create/update and output from create/update/read to avoid contract test failure

*Test*:
```
==================================================================================================================== test session starts =====================================================================================================================
platform darwin -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /Users/anshikg/workspace/cloudformation-cli/env/bin/python3
cachedir: .pytest_cache
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm/.hypothesis/examples')
rootdir: /Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm, configfile: ../../../../../private/var/folders/zh/rhw_046j7bq4d88hjn671hqjy323j8/T/pytest_yxwijb9p.ini
plugins: random-order-1.0.4, localserver-0.5.0, cov-2.10.0, hypothesis-5.19.3, black-0.3.10
collected 16 items                                                                                                                                                                                                                                           

handler_create.py::contract_create_delete PASSED                                                                                                                                                                                                       [  6%]
handler_create.py::contract_invalid_create PASSED                                                                                                                                                                                                      [ 12%]
handler_create.py::contract_create_duplicate PASSED                                                                                                                                                                                                    [ 18%]
handler_create.py::contract_create_read_success PASSED                                                                                                                                                                                                 [ 25%]
handler_create.py::contract_create_list_success PASSED                                                                                                                                                                                                 [ 31%]
handler_delete.py::contract_delete_read PASSED                                                                                                                                                                                                         [ 37%]
handler_delete.py::contract_delete_list PASSED                                                                                                                                                                                                         [ 43%]
handler_delete.py::contract_delete_update PASSED                                                                                                                                                                                                       [ 50%]
handler_delete.py::contract_delete_delete PASSED                                                                                                                                                                                                       [ 56%]
handler_delete.py::contract_delete_create PASSED                                                                                                                                                                                                       [ 62%]
handler_misc.py::contract_check_asserts_work PASSED                                                                                                                                                                                                    [ 68%]
handler_read.py::contract_read_without_create PASSED                                                                                                                                                                                                   [ 75%]
handler_update.py::contract_update_read_success PASSED                                                                                                                                                                                                 [ 81%]
handler_update.py::contract_update_list_success PASSED                                                                                                                                                                                                 [ 87%]
handler_update_invalid.py::contract_update_create_only_property PASSED                                                                                                                                                                                 [ 93%]
handler_update_invalid.py::contract_update_non_existent_resource PASSED                                                                                                                                                                                [100%]

=============================================================================================================== 16 passed in 838.20s (0:13:58) ===============================================================================================================
==================================================================================================================== test session starts =====================================================================================================================
platform darwin -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /Users/anshikg/workspace/cloudformation-cli/env/bin/python3
cachedir: .pytest_cache
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm/.hypothesis/examples')
rootdir: /Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm, configfile: ../../../../../private/var/folders/zh/rhw_046j7bq4d88hjn671hqjy323j8/T/pytest_6r6z6twd.ini
plugins: random-order-1.0.4, localserver-0.5.0, cov-2.10.0, hypothesis-5.19.3, black-0.3.10
collected 16 items                                                                                                                                                                                                                                           

handler_create.py::contract_create_delete PASSED                                                                                                                                                                                                       [  6%]
handler_create.py::contract_invalid_create PASSED                                                                                                                                                                                                      [ 12%]
handler_create.py::contract_create_duplicate PASSED                                                                                                                                                                                                    [ 18%]
handler_create.py::contract_create_read_success PASSED                                                                                                                                                                                                 [ 25%]
handler_create.py::contract_create_list_success PASSED                                                                                                                                                                                                 [ 31%]
handler_delete.py::contract_delete_read PASSED                                                                                                                                                                                                         [ 37%]
handler_delete.py::contract_delete_list PASSED                                                                                                                                                                                                         [ 43%]
handler_delete.py::contract_delete_update PASSED                                                                                                                                                                                                       [ 50%]
handler_delete.py::contract_delete_delete PASSED                                                                                                                                                                                                       [ 56%]
handler_delete.py::contract_delete_create PASSED                                                                                                                                                                                                       [ 62%]
handler_misc.py::contract_check_asserts_work PASSED                                                                                                                                                                                                    [ 68%]
handler_read.py::contract_read_without_create PASSED                                                                                                                                                                                                   [ 75%]
handler_update.py::contract_update_read_success PASSED                                                                                                                                                                                                 [ 81%]
handler_update.py::contract_update_list_success PASSED                                                                                                                                                                                                 [ 87%]
handler_update_invalid.py::contract_update_create_only_property PASSED                                                                                                                                                                                 [ 93%]
handler_update_invalid.py::contract_update_non_existent_resource PASSED                                                                                                                                                                                [100%]

============================================================================================================== 16 passed in 1042.18s (0:17:22) ===============================================================================================================
```


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by prerna-p** [comment]
Is there an open question or ticket that can provide more clarity? Where are readOnly properties being passed to create or update handler? Shouldn't it be fixed there instead?

**Comment by anshikg** [comment]
> Is there an open question or ticket that can provide more clarity? Where are readOnly properties being passed to create or update handler? Shouldn't it be fixed there instead?

I don't have a question for this issue. This happens when a readOnly property can be updated by the service during create/update.

**Review by ammokhov — APPROVED** [review]

File: src/rpdk/core/contract/resource_client.py [hunk 1]
```diff
@@ -82,7 +82,9 @@ def prune_properties_from_model(model, paths):
     The function modifies the document in-place, but also returns the document
     for convenience. (The return value may be ignored.)
     """
-    return prune_properties({"properties": model}, paths)["properties"]
+    for path in paths:
+        model = prune_properties({"properties": model}, path)["properties"]
+    return model
```
Comment (ammokhov) [inline src/rpdk/core/contract/resource_client.py:87]: I would add `try-catch` for `KeyError` exception

**Review by prerna-p — CHANGES_REQUESTED** [review]
+1 for Anton's comment on adding exception handling, LGTM otherwise

**Review by prerna-p — APPROVED** [review]

**Pull request merged**